.github/workflows/docs-check.yml

```diff
@@ -0,0 +1,20 @@
+name: Docs Check
+
+on:
+  pull_request:
+    branches:
+      - master
+    types:
+      - opened
+      - labeled
+      - unlabeled
+
+jobs:
+  fail-if-docs-needed:
+    if: contains(github.event.pull_request.labels.*.name, 'docs-needed')
+    runs-on: ubuntu-latest
+    steps:
+      - name: Fail if PR needs accompanying docs to be written
```

Comment:
Probably you'll prefer this: https://github.com/marketplace/actions/comment-pull-request

Reply:
I feel like a CI check is more relevant - it's not necessary to have an automated comment history saying "please write docs" if they haven't been written yet, and a comment is less of a mental block to merging something in (the point being to make the merger stop and think before doing so if there's a 'failing test').